jobs/metrics-report-generator/src/utils/helpers.util.ts

```diff
@@ -0,0 +1,98 @@
+import { logError, logInfo, logWarn } from '@interop-be-reports/commons'
+import {
+  AgreementsWorksheetTableData,
+  DescriptorsWorksheetTableData,
+  TokensWorksheetTableData,
+} from '../models/excel.model.js'
+import { AgreementQueryData, EServiceQueryData, PurposeQueryData, TenantQueryData } from '../models/query-data.model.js'
+import { randomUUID } from 'crypto'
+import { TokensDataQueryResult } from '../services/athena-queries.service.js'
+
+const cidJob = randomUUID()
+
+export const log = {
+  info: logInfo.bind(null, cidJob),
+  warn: logWarn.bind(null, cidJob),
+  error: logError.bind(null, cidJob),
+}
+
+export function generateAgreementsWorksheetTableData(
+  agreements: AgreementQueryData[],
+  purposes: PurposeQueryData[],
+  eservicesMap: Map<string, EServiceQueryData>,
+  tenantsMap: Map<string, TenantQueryData>
+): AgreementsWorksheetTableData[] {
+  return agreements.map<AgreementsWorksheetTableData>((agreement) => {
+    const agreementPurposes = purposes.filter((purpose) => purpose.eserviceId === agreement.eserviceId)
```

Comment:
if I'm not missing something
```suggestion
    const agreementPurposes = purposes.filter((purpose) => purpose.eserviceId === agreement.eserviceId && purpose.consumerId === agreement.consumerId)
```